src/Raven.Server/Documents/Sharding/ShardedDocumentPutAction.cs

```diff
@@ -22,7 +22,12 @@ protected override void ValidateId(Slice lowerId)
         var bucket = ShardHelper.GetBucketFor(config, lowerId);
         var shard = ShardHelper.GetShardNumberFor(config, bucket);
         if (shard != _documentDatabase.ShardNumber)
+        {
+            if (_documentDatabase.ForTestingPurposes != null && _documentDatabase.ForTestingPurposes.EnableWritesToTheWrongShard)
```

Comment:
In here, shouldn't we trigger the migration checks?

Reply:
Not sure I follow...

Reply:
If we are writing a document that we _know_ isn't right for this shard. We should trigger a check for migration immediately on tx commit.

Reply:
https://issues.hibernatingrhinos.com/issue/RavenDB-20049/Trigger-resharing-if-we-got-a-document-to-the-wrong-shard

Reply:
The problem is that we _don't know_ we are writing to the wrong shard at this point (if we do, we throw `ShardMismatchException`).

Reply:
How can we _not_ know that?
And can't we do this check whenever we:
* Have a shard topology update?